File: Malhar/Task 1/FirstApp/app/src/main/java/com/example/firstapp/MainActivity.java
package com.example.firstapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    Button zero, one, two, three, four, five, six, seven, eight, nine, add, sub, mul, div, ac, eql;

    float v1 = 0 , v2 = 0, sum = 0;
    boolean plus;
    boolean minus;
    boolean multiply;
    boolean divide;
    TextView disp;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        disp = findViewById(R.id.disp);
        zero = findViewById(R.id.zero);
        one = findViewById(R.id.one);
        two = findViewById(R.id.two);
        three = findViewById(R.id.three);
        four = findViewById(R.id.four);
        five = findViewById(R.id.five);
        six = findViewById(R.id.six);
        seven = findViewById(R.id.seven);
        eight = findViewById(R.id.eight);
        nine = findViewById(R.id.nine);
        add = findViewById(R.id.add);
        sub = findViewById(R.id.sub);
        mul = findViewById(R.id.mul);
        div = findViewById(R.id.div);
        ac = findViewById(R.id.ac);
        eql = findViewById(R.id.eql);

        zero.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                disp.setText(disp.getText() + "0");
            }
        });

        one.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                disp.setText(disp.getText() + "1");
            }
        });

        two.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                disp.setText(disp.getText() + "2");
            }
        });

        three.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                disp.setText(disp.getText() + "3");
            }
        });

        four.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                disp.setText(disp.getText() + "4");
            }
        });

        five.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                disp.setText(disp.getText() + "5");
            }
        });

        six.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                disp.setText(disp.getText() + "6");
            }
        });

        seven.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                disp.setText(disp.getText() + "7");
            }
        });

        eight.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                disp.setText(disp.getText() + "8");
            }
        });

        nine.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                disp.setText(disp.getText() + "9");
            }
        });

        add.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(disp.getText().toString().isEmpty())
                    disp.setText("");
                else{
                    if(v2 != 0)
                    {
                        equal();
                        v1 = Float.parseFloat(disp.getText() + "");
                        plus = true;
                    }
                    else
                    {
                        v1 = Float.parseFloat(show.getText() + "");
                        plus = true;
                        v2 = 1;
                    }
                    disp.setText(null);
                }
            }
        });

        sub.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(disp.getText().toString().isEmpty())
                    disp.setText("");
                else{
                    if(v2 != 0)
                    {
                        equal();
                        v1 = Float.parseFloat(disp.getText() + "");
                        minus = true;
                    }
                    else
                    {
                        v1 = Float.parseFloat(show.getText() + "");
                        minus = true;
                        v2 = 1;
                    }
                    disp.setText(null);
                }
            }
        });

        div.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(disp.getText().toString().isEmpty())
                    disp.setText("");
                else{
                    if(v2 != 0)
                    {
                        equal();
                        v1 = Float.parseFloat(disp.getText() + "");
                        divide = true;
                    }
                    else
                    {
                        v1 = Float.parseFloat(show.getText() + "");
                        divide = true;
                        v2 = 1;
                    }
                    disp.setText(null);
                }disp.setText(disp.getText() + "/");
            }
        });

        mul.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(disp.getText().toString().isEmpty())
                    disp.setText("");
                else{
                    if(v2 != 0)
                    {
                        equal();
                        v1 = Float.parseFloat(disp.getText() + "");
                        multiply = true;
                    }
                    else
                    {
                        v1 = Float.parseFloat(show.getText() + "");
                        multiply = true;
                        v2 = 1;
                    }
                    disp.setText(null);
                }
            }
        });

        eql.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(disp.getText().toString().isEmpty()){

                    disp.setText("");
                }
                else{

                    if(sum==0){
                        v2 = Float.parseFloat(disp.getText() + "");
                        if(plus==true){
                            sum = v1 + v2;
                            disp.setText(sum + "");
                        }
                        if(minus == true){
                            sum = v - v2;
                            disp.setText(sum + "");
                        }
                        if(divide == true){
                            sum = v1 / v2;
                            disp.setText(sum + "");
                        }
                        if(multiply == true){
                            sum = v1 * v2;
                            disp.setText(sum + "");
                        }

                    }
                    else{
                        v1=Float.parseFloat(disp.getText() + "");
                        if(plus == true){
                            sum = sum+v1;
                            disp.setText(sum + "");
                        }
                        if(minus == true){
                            sum = sum - v1;
                            disp.setText(sum + "");
                        }
                        if(divide == true){
                            sum = sum / v1;
                            disp.setText(sum + "");
                        }
                        if(multiply == true){
                            sum = v1 * sum;
                            disp.setText(sum + "");
                        }
                    }


                    plus = false;
                    minus = false;
                    divide = false;
                    multiply = false;

                }
            }
        });

        ac.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                disp.setText(null);
                v1 = 0;
                v2 = 0;
                sum = 0;
            }
        });
    }

    public void equal(){
        if(sum==0){
            v2 = Float.parseFloat(disp.getText() + "");
            if(plus == true){
                sum = v1 + v2;
                disp.setText(sum + "");
            }
            if(minus == true){
                sum = v1 - v2;
                disp.setText(sum + "");
            }
            if(divide == true){
                sum = v1 / v2;
                disp.setText(sum + "");
            }
            if(multiply == true){
                sum = v1 * v2;
                disp.setText(sum + "");
            }

        }
        else{
            v1 = Float.parseFloat(disp.getText() + "");
            if(plus == true){
                sum = sum + v1;
                disp.setText(sum + "");
            }
            if(minus == true){
                sum = sum - v1;
                disp.setText(sum + "");
            }
            if(divide == true){
                sum = sum / v1;
                disp.setText(sum + "");
            }
            if(multiply == true){
                sum = v1 * sum;
                disp.setText(sum + "");
            }
        }


        plus = false;
        minus = false;
        divide = false;
        multiply = false;
    }
}

File: Jatin/Task 2/task2/Source_Code/Task2_adg/app/src/main/java/com/example/task20/MainActivity.kt
package com.example.task20

import android.annotation.SuppressLint
import android.app.AlertDialog
import android.app.DatePickerDialog
import android.app.ProgressDialog
import android.app.TimePickerDialog
import android.content.DialogInterface
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.CountDownTimer
import android.os.Handler
import android.view.View
import android.widget.*
import androidx.annotation.RequiresApi
import androidx.core.content.ContextCompat
import com.google.android.material.snackbar.Snackbar
import kotlinx.android.synthetic.main.activity_main.*
import java.text.SimpleDateFormat
import java.util.*

class MainActivity : AppCompatActivity() {
//    var day=0
//    var month=0
//    var year=0
//    var hour=0
//    var minute = 0
//    var savedday=0
//    var savedmonth=0
//    var savedyear=0
//    var savedhour=0
//    var savedminute = 0
//DatePickerDialog.OnDateSetListener,TimePickerDialog.OnTimeSetListener --> have to add this in app compact activity
    var format =SimpleDateFormat("dd MMM,YYYY",Locale.US)
    @RequiresApi(Build.VERSION_CODES.N)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
//        pickDate()

        val snackbar=findViewById<Button>(R.id.snackbar)
        val root_layout=findViewById<androidx.constraintlayout.widget.ConstraintLayout>(R.id.root_layout)
        alert.setOnClickListener {
            val builder= AlertDialog.Builder(this)
            val dialogView=layoutInflater.inflate(R.layout.alert,null)
            builder.setView(dialogView)
            builder.setCancelable(false)
            builder.setPositiveButton("Yes",DialogInterface.OnClickListener(){ dialogInterface: DialogInterface, i->
                finish()
            })
            builder.setNegativeButton("No",DialogInterface.OnClickListener(){ dialogInterface: DialogInterface, i->
                dialogInterface.cancel()
            })
            val alertDialog=builder.create()
            alertDialog.show()
        }
        toast.setOnClickListener {
            Toast.makeText(this,"Hola Amigo!",Toast.LENGTH_LONG).show()
        }
        val handler=Handler()
        progressbar.setOnClickListener{
            var pd =ProgressDialog(this)
            pd.setMessage("Welcome to the app user!")
            pd.setProgressStyle(ProgressDialog.STYLE_SPINNER)
            pd.show()
            var handler=Handler()
            handler.postDelayed(object :Runnable{
                override fun run() {
                    pd.dismiss()
                }

            },5000)

        }



        Date.setOnClickListener {
            val now =Calendar.getInstance()
            val datepicker=DatePickerDialog(this,DatePickerDialog.OnDateSetListener{view,year,month,dayOfMonth->
                val selectDate=Calendar.getInstance()
                selectDate.set(Calendar.YEAR,year)
                selectDate.set(Calendar.MONTH,month)
                selectDate.set(Calendar.DAY_OF_MONTH,dayOfMonth)
                val date =format.format(selectDate.time)
                Date.text="Date: $date"

            },
                now.get(Calendar.YEAR),now.get(Calendar.MONTH),now.get(Calendar.DAY_OF_MONTH))
            datepicker.show()

        }

        snackbar.setOnClickListener {
            val snackbar = Snackbar.make(root_layout,"Welcome to the app", Snackbar.LENGTH_INDEFINITE)
            snackbar.setAction("Close", View.OnClickListener {
                snackbar.dismiss()
            })

            snackbar.setActionTextColor(ContextCompat.getColor(this,R.color.nayablue))
            snackbar.show()



        }
    }
    // This is for when we want date and time both

//    private fun getDateTimeCalender(){
//        val cal= Calendar.getInstance()
//        day=cal.get(Calendar.DAY_OF_MONTH)
//        month=cal.get(Calendar.MONTH)
//        year=cal.get(Calendar.YEAR)
//        hour =cal.get(Calendar.HOUR)
//        minute=cal.get(Calendar.MINUTE)
//    }
//    private fun pickDate(){
//
//        date.setOnClickListener {
//            getDateTimeCalender()
//            DatePickerDialog(this,this,year,month,day).show()
//        }
//
//    }
//
//    override fun onDateSet(view: DatePicker?, year: Int, month: Int, dayOfMonth: Int) {
//        savedday=dayOfMonth
//        savedmonth=month
//        savedyear=year
//        getDateTimeCalender()
//        TimePickerDialog(this,this,hour, minute,true).show()
//    }
//
//    @SuppressLint("SetTextI18n")
//    override fun onTimeSet(view: TimePicker?, hourOfDay: Int, minute: Int) {
//        savedminute=minute
//        savedhour=hourOfDay
//        date.text="$savedhour:$savedminute\n$savedday/$savedmonth/$savedyear"
//
//
//    }

}